SimPEG/electromagnetics/time_domain/simulation.py

```diff
@@ -86,36 +100,79 @@ def fields(self, m):
         :return f: fields object
         """
 
-        tic = time.time()
+        # Update current model
         self.model = m
 
+        # Instantiate the fields object and pair with the simulation
         f = self.fieldsPair(self)
 
-        # set initial fields
-        f[:, self._fieldType + "Solution", 0] = self.getInitialFields()
+        # Compute initial fields for all sources
+        f[:, self._fieldType + "Solution", 0] = self.getInitialFields()  # mesh x n_src
+
+        # Store Factorizations
+        if self.forward_only is False:
+
+            # Clean factorizations for preexisting model.
+            if hasattr(self, "Ainv"):
+                {k: v.clean() for k, v in self.Ainv.items()}
+            else:
+                self.time_steps = self.dt_threshold * np.round(
```

Comment:
This will fail very quickly if dt_threshold is 0

Reply:
K. So let's update the setter so that dt_threshold must be larger than 0. That work?